src/main/scala/io/iohk/ethereum/db/storage/CachedKeyValueStorage.scala

```diff
@@ -0,0 +1,39 @@
+package io.iohk.ethereum.db.storage
+
+import io.iohk.ethereum.common.SimpleMap
+import io.iohk.ethereum.db.cache.Cache
+import scala.collection._
+
+trait CachedKeyValueStorage[K, V, T <: CachedKeyValueStorage[K, V, T]] extends SimpleMap[K, V, T]{
+  type I <: KeyValueStorage[K, V, I]
+  protected val storage: I
+  protected val cache: Cache[K, V]
+  def apply(cache: Cache[K,V] , storage: I): T
+
+  def get(key: K): Option[V] = cache.get(key) orElse storage.get(key)
```

Comment:
Do you think it could improve performance if we store in the cache everything we read? If so, maybe we can leave a comment here for a follow up task

Reply:
hmm I tested this idea by creating an implementation of `Cache[K,V]` which cached all data in `Scaffeine` cache, and there weren't big performance gains. ( although maybe `todo` is worth adding as this was quick check so i could missed something out)